.github/workflows/fidesops_safe_pr_checks.yml

```diff
@@ -0,0 +1,239 @@
+name: Fidesops CI Checks - Safe
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  CONTAINER: fidesops-local
+  IMAGE: ethyca/fidesops:local
+
+jobs:
+  Build:
```

Comment:
5 minutes? that seems kind of high, but also I don't think 5 minutes should make a huge difference here, CI latency shouldn't be _too_ important and this method feels more efficient. If people are debugging in CI that's an anti-pattern and we should figure out the cause and fix it in my opinion

Reply:
I was surprised by the 5 minutes, I was expecting they way you did it here to be faster. But I was consistently seeing it take about that much longer.

Reply:
The isort job took 51 seconds total, and the initial build took ~5 minutes. seems fine to me?

Reply:
Well, initial build takes around 8 minutes which adds about 5 minutes total because the other stages are faster, they just took longer to get started, but that is splitting hairs.

Really it's just that there is 5is minutes of unneeded wait time before tests start. Looks like the plan in #990 is to make the static checks run with nox outside of docker. At that point the the static checks become a non issue and the only question remains with the other tests (unit, integration). There the test suite goes from taking around 10 minutes to around 15 minutes because of waiting for the build before starting the tests.

Reply:
Just to clarify my real question here: is there a benefit to sharing the build that I am not seeing that makes it worth it to add the extra time to CI? I wouldn't want to add time to CI if there isn't a good reason, but if there is a benefit that would be different.

Reply:
The only real benefit here is minimizing CPU time in our CI pipelines. From an actual cost perspective is it actually impactful? No, not really, CI time is cheap and when we halve the number of repos it'll matter even less.

_but_...I'd like to think that CI runtime isn't very latency sensitive (to a point) and at the very least we're saving the environment a bit lol

Shouldn't the developer know the outcome of the CI checks before they even run?

Reply:
To clarify, I'm not at all willing to die on this hill, just seems wasteful somehow to build the container 10 times in a single CI run

Reply:
I looked at some new bench marks and think we can both be happy here. By removing the build from the docker compose file it looks to have sped up bringing the containers up enough to negate the benefits of the separate builds making total CI times roughly equal. That removes my only issue so let’s go with it.